### ISSUE TITLE
.not() doesn't fire, and when it does it does so multiple times

### ISSUE THREAD

**Issue by d3x0r** [amark/gun#199]
something with local storage only.....

```
var db={};
db.localDb = Gun( {prefix: 'local/'}).get( "org.example.application.chat_service" );
db.local = db.localDb.path( "local_user" )
( db.player.identity = 
db.player.local.path("user_identity") ).not( doDefaultInit ).val( receiveDefaultValue );
```

doDefaultInit does not get called.  For a workaround I added a timer which would insert a value somewhere in the local database which then will trigger the not.

```
var defaultTimeout = setTimeout( function(){ 
                                           db.player.local.path( "init" ).put( true );  
                                      }, 250 );

function receiveDefaultValue() { /*... */ 
       clearTimeout( defaultTimeout );       
 }
function doDefaultInit() {
    db.player.identity.put( db.player.id )
    db.player.local.path("default_room").put( "default_room" )
}
```

and at this point doDefaultInit (the not callback ) fires twice.  (it fires many more times if in a connected database)

---

While testing at first exact results were not matched, so there's also some configuration that needs to be setup also?  Does the globalDb connection also need to be shown?

**Comment by amark** [comment]
@d3x0r thanks for getting this reported.

Do the workarounds allow you to continue developing the game? In which case I'm going to put the milestone for this as v0.4.x/v0.5.x .

However, if this problem is barring further development for you I will prioritize it.

Thanks for reporting!

**Comment by d3x0r** [comment]
Well... I added like 'var notXXXProcessed = false;'  and set it to true to block the multiple .not().

Eventually though I wrote something like .not() .val ()  and neither fires, while I do have data in the tree, and it's localhost connected... 
And now that the data exists, the kicks don't wake it up.
```
 db.world.voxelInfo = db.world.db.path( "voxelInfo" )
 db.world.voxelInfo.map().val( loadVoxels );
```
after having some loaded never gets called?
so separating .not.val into .map.val and a separate not, the val still doesn't go?

**Comment by amark** [comment]
@d3x0r that is frustrating! I'm sorry, is that causing everything else to break/fail?

Mind giving me a solid "yes/no" on whether I can leave this to be fixed with 0.5/0.4? If it is stopping/halting/preventing further development for you, then I'd we should prioritize it.

**Comment by d3x0r** [comment]
It shouldn't be, but apparently is blocking?
I mean if I added the work arounds, and now there's day, either one tiny thing is still unset because of this, or... there's something simple I'm overlooking.
There's also lots of things I can do around the issue without involving the database... and I do have my heart set on getting the database nearly working and that desire will keep me from doing anything else...
but it's not going anywhere; there's no deadlines... I can work on actually making sectors work; and then when they have code saving their state....err I guess that's more database requirement :(
.... well there IS webAudio to figure out...
So ya, when it's ready (in 0.5) I'll gladly test :) 
I am a 'follow the bleeding edge' ... side note?  Know the difference between cutting edge and bleeding edge?  The bleeding edge is just a little ahead :)

**Comment by amark** [comment]
hahaha @d3x0r love it. Sorry I'm causing you bleeding pain ;) and I'll let you know as 0.5 progresses so you can finally not be hurting. :P Thanks for the patience!

**Comment by amark** [comment]
This got fixed, right? Because 0.5 finally went out. So I'm gonna close :).

BTW, @d3x0r how are you doing? How is life going? I hope you are doing well!
